### PR TITLE
Update `clap` to 4.1

### PR DESCRIPTION
**Pull request by nickelc**
This PR is a continuation of #1253 

Thanks to @tjquillan

### CODE REVIEW

File: src/options/set.rs [hunk 1]
```diff
@@ -88,7 +89,7 @@ pub fn set_options(
 
     // --color-only is used for interactive.diffFilter (git add -p) and side-by-side cannot be used
     // there (does not emit lines in 1-1 correspondence with raw git output). See #274.
-    if config::user_supplied_option("color-only", arg_matches) {
+    if config::user_supplied_option("color_only", arg_matches) {
```
Comment (dandavison) [inline src/options/set.rs:92]: How sure are we that we've updated all places to snake-case that need updating?

Comment (nickelc) [reply src/options/set.rs:92]: All usage of the `ArgMatches` seems to go through the `options::set::set_options` function, the `set_options!` macro and `config::user_supplied_option` function.
And the macro checks the names against the `expected_option_name_map` hash map.

**Comment by dandavison** [comment]
It looks like we've lost the colours from `delta -h` and `delta --help`. I liked them, but there was a drawback which was that the ANSI sequences remain even when the help text was being piped/redirected, so e.g. if redirecting to less, the user had to know to use `less -R`.

WDYT? ANy chance clap v4 has made the colors go away when stdout is not a tty?

**Comment by nickelc** [comment]
> Any chance clap v4 has made the colors go away when stdout is not a tty?

Clap's v4 release blog post explains the reasons for this change and plans for customization.
https://epage.github.io/blog/2022/09/clap4/#polishing-help-output

**Comment by dandavison** [comment]
I see, thanks for explaining. Really useful work here, thanks.

**Pull request merged**